executor/src/witgen/identity_processor.rs

```diff
@@ -113,6 +116,41 @@ impl<'a, 'b, T: FieldElement> IdentityProcessor<'a, 'b, T> {
         unimplemented!("No executor machine matched identity `{identity}`")
     }
 
+    // This would be convenient, but doesn't compile.
+    // pub fn iter_machines<'c>(
```

Comment:
Not sure why this doesn't work

Reply:
Also, of course, ideally, this could be a method (using `&mut self`). I think that introduces more issues though, because `self` is then mutably borrowed for as the return value lives.